src/App.tsx

```diff
@@ -1,24 +1,278 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  addTodo,
+  getTodos,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { FilterValues } from './constants';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6438;
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [errorType, setErrorType] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+  const [removingTodoIds, setRemovingTodoIds] = useState<number[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(FilterValues.ALL);
+  const activeTodos = [...todos].filter(todoItem => !todoItem.completed);
+  const isEveryCompleted = todos.every(todoItem => todoItem.completed);
+  const hasCompletedTodo = todos.some(todo => todo.completed);
+
+  const getTodosFromServer = async () => {
+    try {
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch (error) {
+      setHasError(true);
+      setErrorType('upload');
+    }
+  };
+
+  const handleAddTodo = useCallback(async () => {
+    const newTodoToFetch = {
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    const newTodoToShow = {
+      ...newTodoToFetch,
+      id: 0,
+    };
+
+    setIsTodoAdding(true);
+    setTempTodo(newTodoToShow);
+
+    try {
+      const addedTodo = await addTodo(newTodoToFetch);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+
+      setHasError(false);
+    } catch (error) {
+      setHasError(true);
+      setErrorType('add');
```

Comment:
create an enum to keep all errors organized
```suggestion
      setErrorType(ErrorTypes.Add);
```